### PR TITLE
feat: checkbox and radio pseudo elements

### PR DESCRIPTION
**Pull request by JKMarkowski**
## Related Issue
Closes https://github.com/SAP/fundamental-styles/issues/515

## Description
Because of IE11, there was no way to add modify `:after` and `::before` pseudo elements inside input[checkbox/radio]. So now pseudo elements belongs to labels. 
The structure of inputs are changed now, before it was: 
```html
<label class="fd-form-label fd-form-label--radio" for="pDidh761">
    <input type="radio" class="fd-radio" id="pDidh761" name="radio">Field label
</label>
```
And after:
```html
<input type="radio" class="fd-radio" id="pDidh761" name="radio">
<label class="fd-radio__label" for="pDidh761">
    Field label
</label>
```

The `id` and `for` attributes are mandatory since now, because it's the only way we cah connect label with input.

### CODE REVIEW

File: src/checkbox.scss [hunk 1]
```diff
@@ -12,195 +12,134 @@ $fd-checkbox-dimensions-compact: 1rem;
...
+      display: inline-flex;
+      justify-content: center;
+      align-items: center;
+      font-family: "SAP-icons";
+      background-color: var(--sapField_Background);
+      margin: $fd-checkbox-margin;
+      color: var(--sapSelectedColor);
+      border: var(--sapField_BorderWidth) solid var(--sapField_BorderColor);
+      border-radius: var(--sapField_BorderCornerRadius);
+      font-style: normal;
```
Comment (salarenko) [inline src/checkbox.scss]: Setting default values?
`font-style: normal;`
`font-weight: normal;`

File: src/mixins/_forms.scss [hunk 2]
```diff
@@ -139,19 +139,35 @@
...
+  }
+}
 
-    box-shadow: none;
-    outline: none;
+@mixin fd-form-label() {
+  @include fd-reset();
+  @include fd-ellipsis();
+
+  width: fit-content;
```
Comment (salarenko) [inline src/mixins/_forms.scss]: `width: fit-content;` is not supported in IE11 and Edge.
If label is always going to be flex child I recommend to use `align-self: flex-start`

File: src/radio.scss [hunk 3]
```diff
@@ -15,147 +15,122 @@ $fd-radio-outer-circle-margin-compact: 0.5rem;
...
-      width: $fd-radio-inner-circle-diameter-compact;
-      height: $fd-radio-inner-circle-diameter-compact;
+  @extend %fd-radio-input-hidden;
+
+  &__label {
+    @include fd-form-label();
+
+    display: flex;
+    align-items: center;
+    cursor: pointer;
```
Comment (salarenko) [inline src/radio.scss]: Be consequent, use `@include action-cursor();` or `cursor: pointer;`, not both

File: src/radio.scss [hunk 4]
```diff
@@ -15,147 +15,122 @@ $fd-radio-outer-circle-margin-compact: 0.5rem;
...
+      display: inline-flex;
+      justify-content: center;
+      align-items: center;
+      font-family: "SAP-icons";
+      background-color: var(--sapField_Background);
+      margin: $fd-radio-outer-circle-margin;
+      color: var(--sapSelectedColor);
+      border: var(--sapField_BorderWidth) solid var(--sapField_BorderColor);
+      border-radius: 50%;
+      font-style: normal;
```
Comment (salarenko) [inline src/radio.scss]: Default values?

**Comment by JKMarkowski** [comment]
Thanks @salarenko for review, I added some changes due to your comments.

**Comment by stefanoScalzo** [comment]
We have an issue for dark mode with the checkboxes
Screen Shot 2019-12-19 at 10 51 53 AM

File: src/radio.scss [hunk 5]
```diff
@@ -7,155 +7,128 @@
...
+    @include action-cursor();
+
+    display: flex;
+    align-items: center;
+
+    &::before {
+      content: "";
+      height: $fd-radio-outer-circle-diameter;
+      width: $fd-radio-outer-circle-diameter;
+      font-size: 0.6875rem;
```
Comment (InnaAtanasova) [inline src/radio.scss]: Shouldn't this be `$fd-radio-inner-circle-diameter` ?

File: src/radio.scss [hunk 6]
```diff
@@ -7,155 +7,128 @@
...
+    + .#{$block}__label {
+      pointer-events: none;
+      opacity: 0.4;
+    }
+  }
+
+  &--compact {
+    + .#{$block}__label {
+      &::before {
+        font-size: 0.4375rem;
```
Comment (InnaAtanasova) [inline src/radio.scss]: Maybe `$fd-radio-inner-circle-diameter-compact` ?

File: src/checkbox.scss [hunk 7]
```diff
@@ -7,200 +7,136 @@
...
 
-    &:indeterminate {
-      &::after {
-        background-color: var(--sapField_SuccessColor);
-      }
-    }
+  &:indeterminate + .#{$block}__label::before {
+    padding-bottom: 0.0625rem;
+    content: "\e17b";
+    font-size: 0.875rem;
```
Comment (InnaAtanasova) [inline src/checkbox.scss]: It should be 0.75 x 0.75 rem sized square

File: src/checkbox.scss [hunk 8]
```diff
@@ -7,200 +7,136 @@
...
 
-    @include fd-hover() {
-      @include fd-disabled() {
-        border-color: var(--sapField_SuccessColor);
-      }
-    }
+  &:indeterminate + .#{$block}__label--compact::before {
+    content: "\e17b";
+    padding-bottom: 0;
+    font-size: 0.5625rem;
```
Comment (InnaAtanasova) [inline src/checkbox.scss]: It should be 0.5 x 0.5 rem

**Comment by InnaAtanasova** [comment]
There's padding-right and margin-right 
Screen Shot 2019-12-19 at 4 39 28 PM
Screen Shot 2019-12-19 at 4 39 10 PM
Screen Shot 2019-12-19 at 4 33 47 PM
Screen Shot 2019-12-19 at 4 20 11 PM

**Comment by InnaAtanasova** [comment]
Fieldset legend has a pointer cursor on hover

File: src/checkbox.scss (hunk 7 above)
Comment (JKMarkowski) [reply src/checkbox.scss]: It should be and it is. In general `font-size: 0.75rem` doesn't look good. It looks same as before, when we add 0.75rem + 0.0625rem (1px)

File: src/checkbox.scss (hunk 8 above)
Comment (JKMarkowski) [reply src/checkbox.scss]: As above, 0.5rem doesn't look good.

**Comment by JKMarkowski** [comment]
Hello @InnaAtanasova padding is removed now, there i only margin left. Legend cursor is auto, instead of pointer.

**Comment by stefanoScalzo** [comment]
Screen Shot 2019-12-20 at 1 16 52 PM
The focus must be on the label and not full width

**Comment by InnaAtanasova** [comment]
> Hello @InnaAtanasova padding is removed now, there i only margin left. Legend cursor is auto, instead of pointer. IE11 offset is fixed.

Why do we need the margin? A few months ago a team reported this as an issue. It's very noticeable when there's no label text 
Screen Shot 2019-12-20 at 1 50 00 PM

**Comment by InnaAtanasova** [comment]
The Breaking changes need to be documented

**Pull request merged**